tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/ZipUtil.java

```diff
@@ -0,0 +1,66 @@
+package com.linagora.tmail.james.jmap;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.jmap.routes.Blob;
+
+import reactor.core.publisher.Flux;
+
+public class ZipUtil {
+    public static Flux<byte[]> createZipStream(Collection<Blob> blobs) {
+        return Flux.create(sink -> {
+            try (ZipOutputStream zos = new ZipOutputStream(new OutputStream() {
+                @Override
+                public void write(int b) {
+                    sink.next(new byte[]{(byte) b});
+                }
+
+                @Override
+                public void write(byte[] b, int off, int len) {
+                    sink.next(java.util.Arrays.copyOfRange(b, off, off + len));
+                }
+            })) {
+                for (Blob blob: blobs) {
+                    zos.putNextEntry(new ZipEntry(blob.blobId().toString()));
+                    byte[] buffer = new byte[4096];
+                    int len;
+                    while ((len = blob.content().read(buffer)) > 0) {
+                        zos.write(buffer, 0, len);
+                    }
+                    zos.closeEntry();
+                }
+                zos.finish();
+                sink.complete();
+            } catch (Exception e) {
+                sink.error(e);
+            }
+        });
+    }
+
+    public static String readZipData(InputStream inputStream) {
+        StringBuilder result = new StringBuilder();
+
+        try (ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+            ZipEntry entry;
+            while ((entry = zipInputStream.getNextEntry()) != null) {
+                if (!entry.isDirectory()) {
+                    result.append("File: ").append(entry.getName()).append("\n");
+                    String fileContent = IOUtils.toString(zipInputStream, StandardCharsets.UTF_8);
+                    result.append(fileContent).append("\n");
+                }
+                zipInputStream.closeEntry();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        return result.toString();
+    }
```

Comment:
Won't a return type being a `List<JMAPZipEntry>`

With

```
record JMAPZipEntry(String filename, String content)
```

Better describe the underlying data?